src/data_types/link_secret.rs

```diff
@@ -46,36 +36,28 @@ impl TryInto<MasterSecret> for LinkSecret {
     }
 }
 
-impl TryInto<MasterSecret> for &LinkSecret {
+impl TryInto<ursa::bn::BigNumber> for LinkSecret {
     type Error = ConversionError;
 
-    fn try_into(self) -> Result<MasterSecret, Self::Error> {
-        let j = serde_json::json!({
-            "ms": self.0
-        });
-
-        serde_json::from_value(j)
-            .map_err(|err| ConversionError::from_msg(format!("Error creating link secret: {err}")))
+    fn try_into(self) -> Result<ursa::bn::BigNumber, Self::Error> {
+        let b = self.0.to_be_bytes();
+        ursa::bn::BigNumber::from_bytes(&b).map_err(|err| {
+            ConversionError::from_msg(format!(
+                "Error transforming link secret into bignumber: {err}"
+            ))
+        })
     }
 }
 
-impl TryInto<String> for LinkSecret {
-    type Error = ConversionError;
-
-    fn try_into(self) -> Result<String, Self::Error> {
-        self.0
-            .to_dec()
-            .map_err(|err| ConversionError::from_msg(format!("Error creating link secret: {err}")))
+impl Into<String> for LinkSecret {
+    fn into(self) -> String {
+        self.0.to_string()
     }
 }
 
-impl TryFrom<&str> for LinkSecret {
-    type Error = ConversionError;
-
-    fn try_from(value: &str) -> Result<Self, Self::Error> {
-        Ok(Self(BigNumber::from_dec(value).map_err(|err| {
-            ConversionError::from_msg(format!("Error creating link secret: {err}"))
-        })?))
+impl From<&str> for LinkSecret {
+    fn from(value: &str) -> Self {
+        Self(U256::from_be_hex(value))
```

Comment:
This will panic on bad input, which doesn't seem ideal. I've been decoding the hex to bytes and using U256::from_be_slice but crypto-bigint could use fallible methods for this.